driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SingleResultSubscriber.java

```diff
@@ -33,7 +33,7 @@
     @Nullable
     T get() {
         try {
-            if (!latch.await(5, TimeUnit.SECONDS)) {
+            if (!latch.await(30, TimeUnit.SECONDS)) {
```

Comment:
This troubles me.  I wonder what we're waiting for, and why it happens after 30 seconds but not 5. What do you think?

Reply:
This is mainly to catch slow servers / replicaset primary changes that cause errors and take longer than 5 seconds to complete.

Reply:
We shouldn't get failovers on Evergreen, but I guess it can happen in extreme circumstances.